Adding a test to people nearby search method and sort people by recent signups  [WIP]

Nada1996 opened the pull request:
Fixes #3729  #3766  (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

Nada1996 commented:
@jywarren @milaaraujo I added a test to the tagNearbyPeople function, also it now sorts the results by the most recent signups by default.

Now, I want to pass the "order_by" as a param to sort it by followers or recent logins for example, but  I want some feedback. 
Thanks in advance

Nada1996 commented:
@milaaraujo Thanks for your feedback, I working on it now adjusting the code then I'll push the code to the PR right away 👍

milaaraujo commented:
Great, `tagNearbyPeople()` method looks good!

Now, about the tests, you still need to do the assertions to compare what you were expecting and what the endpoint really returns. Take  look at [this test for example](https://github.com/publiclab/plots2/blob/master/test/functional/api/search_api_test.rb#L11-L33):

>assert_equal "/profile/steff3",   json['items'][0]['doc_url']
assert_equal "/profile/steff2",   json['items'][1]['doc_url']
assert_equal "/profile/steff1", json['items'][2]['doc_url']

You will need to use the data in https://github.com/publiclab/plots2/tree/master/test/fixtures. Let me know if you need any help with this.

Nada1996 commented:
@milaaraujo I added locations tags to existing users and adjusted the code accordingly,
I also added some asserts to check on the sorting functionality and the search by tag.
Please let me know if there's anything else that needs to be done.
Thanks in advance

milaaraujo commented:
This is looking great! I just added a small review. Thank you for your work here. :smile:

What do you think about it, @jywarren?

jywarren commented on app/services/search_service.rb:
```diff
@@ -135,14 +135,19 @@ def tagNearbyPeople(query, tag, limit = 10)
     end
 
-    items = items.limit(limit)
+    items = items.order(id: :desc).limit(limit)
```
This is fantastic. It's really readable and well implemented, and with a great test. Just this one issue to resolve and it's ready to merge!

milaaraujo commented:
Thank you, @Nada1996!

jywarren commented:
This looks super. Fantastic work, and great teamwork from @milaaraujo as well!
